code/modules/goonchat/browserOutput.dm

```diff
@@ -172,6 +177,18 @@ GLOBAL_DATUM_INIT(iconCache, /savefile, new("tmp/iconCache.sav")) //Cache of ico
 
 //Called by client, sent data to investigate (cookie history so far)
 /datum/chatOutput/proc/analyzeClientData(cookie = "")
+	//Spam check
+	if(world.time  >  last_check + (3 SECONDS))
+		last_check = world.time
```

Comment:
If you want to optimize this a tiny bit you could do, not that it matters too much but this is a performance pr
```
if(world.time > expired_check)
  total_checks = 0
expired_check = world.time + 3 SECONDS
```

Reply:
Also I think it was a mistake that you don't refresh the check time when it's not passed?

Reply:
we only want to extend the check period after we have reset the total_checks counter